### PR TITLE
feat: add ability to specify iframe script origin and iframe notify parent for silent auth

### PR DESCRIPTION
**Pull request by pavliy**
**The problem**
Both - oidc-client-js & oidc-client-ts - support silent flow only within the same domain. However in my case - apps are hosted on different domains. It's possible to address it by specifying 2 new settings which will allow to overcome issue and make all things work even with different domains.

**What for**
I'm working on micro frontends implementation currently. 
- Architecture is based on webpack module federation and web components
- There is a shell hosted on on https://app-a.test.com
- And there is micro frontend app loaded through the shell, hosted as https://app-b.test.com
- Shell and each micro app - are separate relying parties which have their own client id, redirect urls etc.  Users can also access them directly.

The main goal is to avoid explicit login flow for getting micro app access token (user does not need to enter credentials once again)
Though, both apps are separate relying parties, token is issued by the same authority.

So the main idea:
- User logs in to shell: https://app-a.test.com and token is saved into session storage
- User clicks the link to load micro app - code is loaded using module federation, which includes oidc-client load dependency as well
- Silent auth flow happens to get token for micro app. And in result - it's also saved in session storage 

BTW, I'll add unit tests soon to cover updated scenarios
### Checklist

- [x] This PR makes changes to the public API 
- [ ] I have included links for closing relevant issue numbers

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "oidc-client-ts",
-  "version": "2.0.3",
```
Comment (pamapa) [inline package.json:3]: Please do not touch package-lock.json and package.json in your MR.

File: .gitignore [hunk 2]
```diff
@@ -21,3 +21,4 @@ node_modules/
 .DS_Store
 .vscode/
 temp/
+.history/
```
Comment (pamapa) [inline .gitignore:24]: i do not see why this would be needed

Comment (pavliy) [reply .gitignore:24]: I'm using localhistory plugin for vscode (https://marketplace.visualstudio.com/items?itemName=xyz.local-history)
So that's just to ignore files generated by it.

File: package.json [hunk 3]
```diff
@@ -37,6 +37,7 @@
     "clean": "git clean -fdX dist lib *.tsbuildinfo",
     "prepack": "npm run build",
     "test": "tsc && jest",
+    "test:watch": "npm run test -- --watch",
```
Comment (pamapa) [inline package.json:40]: if you want this please make another MR with only that

**Comment by pamapa** [comment]
Please rebase all commits into one commit (as from different users...). Otherwise LGTM.

@kherock Can you have a look too?

**Comment by pavliy** [comment]
> Please rebase all commits into one commit (as from different users...). Otherwise LGTM.
> 
Will do. Thnx

**Comment by pavliy** [comment]
@pamapa @kherock please let me know if anything else is needed in this MR.
thnx

**Comment by pamapa** [comment]
i have added one comment to fix, if this is done i will merge this unless @kherock as more to add...

File: src/UserManagerSettings.ts [hunk 4]
```diff
@@ -36,6 +36,10 @@ export interface UserManagerSettings extends OidcClientSettings {
     /** The methods window.location method used to redirect (default: "assign") */
     redirectMethod?: "replace" | "assign";
 
+    /** The target to pass while calling postMessage inside iframe for callback */
```
[the remark this answers is not in the document]
Comment (pavliy) [reply src/UserManagerSettings.ts]: Makes sense. thnx.

**Review by pamapa — APPROVED** [review]

**Comment by pamapa** [comment]
great work, thanks for contributing
